network-api/networkapi/wagtailpages/pagemodels/base.py

```diff
@@ -33,6 +36,37 @@ class BasePage(FoundationMetadataPageMixin, FoundationNavigationPageMixin, Page)
     class Meta:
         abstract = True
 
+    def get_donate_banner(self, request):
+        # Check if theres an active A/B test for the homepage.
+        homepage = self.get_ancestors().type(Homepage).specific().first()
+        active_ab_test = AbTest.objects.filter(page=homepage, status=AbTest.STATUS_RUNNING).first()
+
+        # If theres no A/B test found, return the donate_banner field as usual.
+        if not active_ab_test:
+            return homepage.donate_banner
+
+        # Check for the cookie specific to this A/B test.
+        test_cookie_name = f"wagtail-ab-testing_{active_ab_test.id}_version"
```

Comment:
In the `wagtail-ab-testing` source code, the cookie name is defined as "`wagtail-ab-testing_{ab_test.id}_version`".
For reference, see the relevant source code [here](https://github.com/wagtail-nest/wagtail-ab-testing/blob/main/wagtail_ab_testing/wagtail_hooks.py#L196-L197).